recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -14,11 +14,12 @@ class PahoMqttcConan(ConanFile):
     options = {"shared": [True, False],
                "fPIC": [True, False],
                "ssl": [True, False],
-               "asynchronous": [True, False]}
-    default_options = {"shared": False,
+               "samples": [True, False]}
+    # static builds didn't really work until 1.3.4
+    default_options = {"shared": True,
```

Comment:
Please don't change this. Statis should be the default for all recipes who support static at all

Reply:
Static didn't work properly until 1.3.4 as that was never the intention when the library was first written in 2008.

Reply:
Okay, but then please only change the default to shared for < 1.3.4

Reply:
I haven't found a way to have version specific defaults.  One way would be to remove the older versions of the library, in due course.

Reply:
What do you mean by don't work properly? Paho's recipe is not new, there is an older recipe in Conan Community and we didn't receive any issue about. Do you have any evidence, like upstream issues?

Reply:
I think he was talking about the windows variant regarding not working correctly as static.  I have been using the library for a couple years now and it is true that while static will indeed pass a build on Windows it won't run properly and I had to force my downstream recipes and packaging to shared for Windows builds.  Ian is the original author of this library so I think it is best we accept his advice on using it and move forward with this much needed set of fixes.

Reply:
That was one reason for creating a separate recipe for 1.3.4 - the default for the shared option could be different for 1.3.4 and the previous versions.  In any case, I have reverted to one recipe now, as requested, and set the default for all for the time being, to shared.

Reply:
Hey Ian, I owe you a beer! I used paho mqtt c for a couple projects in the past!

If it's possible we such configuration (static) we can drop by throwing a ConanInvalidConfiguration and using shared by default. WDYT ?

Reply:
@uilianries Cool!  As we're not likely to meet, if you want to buy me a beer you could sponsor me on Github for a month or two for a couple of $ :-)

Throwing an invalid configuration for windows static builds < 1.3.4 sounds like a good option.  I'll look into it.

I've set the ssl option default to True because just about everyone will want encryption, and I've fixed the SSL pre 1.3.4 builds in a patch.

Also, I think the async builds are more popular than the other variant so that should be the default.  However, the MQTTAsync and MQTTClient libraries are actually different APIs, so I was wondering if they should be in different recipes entirely?

Reply:
Both can be used on same recipe, because we use 1 project x 1 recipe. I understand your point, looking as a developer, I can be interested on only one of them (async, for instance), but in terms of size and option is too small to get it separated.

Reply:
I agree most people probably use the async api and library but a good optimizing linker should optimize out unused functions if linked statically so it probably makes sense to just leave them in the same recipe as long as it doesn't cause conan/cmake confusion in linking against both libs in the same package.  Is it possible to just output a single .lib with both interfaces always might be simpler to deal with in general.